### ISSUE TITLE
react: Attachments in afterhooks are not rendered

### ISSUE THREAD

**Issue by mpkorstanje** [cucumber/cucumber#995]
## Summary

Attachments attached in an after hook are not rendered in the html-report. While images attached as part of a step are. This is esp annoying because people use after hooks to attach screenshots when the scenario failed.

To reproduce: Use the cck to attach an image in an after hook. Observe that it is not included in the html report.

```
    @After
    public void aJPEGImageIsAttachedInAfterHook() throws IOException {
        Path path = Paths.get("src/test/resources/features/attachments/cucumber-growing-on-vine.jpg");
        byte[] bytes = Files.readAllBytes(path);
        String fileName = path.getFileName().toString();
        scenario.attach(bytes, "image/jpg", fileName);
    }
```

Use the cck to attach an image in a step. Observe that it is included in the html report.

```
    @When("a JPEG image is attached")
    public void aJPEGImageIsAttached() throws IOException {
        Path path = Paths.get("src/test/resources/features/attachments/cucumber-growing-on-vine.jpg");
        byte[] bytes = Files.readAllBytes(path);
        String fileName = path.getFileName().toString();
        scenario.attach(bytes, "image/jpg", fileName);
    }
```

**Comment by vincent-psarga** [comment]
@bastienmalatier il working on it, see the PR #975  :)

**Comment by mpkorstanje** [comment]
Ah. I didn't realize its all related.

**Comment by mpkorstanje** [comment]
@vincent-psarga I'm still having some problems getting this to work with Cucumber JVM. 

I did find a few bugs in how JVM assigned the ids, but what I have here should be correct. Any idea what is wrong here?

```
{"meta":{"protocolVersion":"12.1.1","implementation":{"name":"cucumber-jvm","version":"6.1.3-SNAPSHOT"},"runtime":{"name":"Ubuntu","version":"11.0.7"},"os":{"name":"Linux"},"cpu":{"name":"amd64"}}}
{"testRunStarted":{"timestamp":{"seconds":"1593102300","nanos":609281000}}}
{"source":{"uri":"classpath:io/cucumber/examples/java/basic_arithmetic.feature","data":"@foo\nFeature: Basic Arithmetic\n\n  Scenario: A Calculator\n    Given a calculator I just turned on\n\n","mediaType":"text/x.cucumber.gherkin+plain"}}
{"gherkinDocument":{"uri":"classpath:io/cucumber/examples/java/basic_arithmetic.feature","feature":{"location":{"line":2,"column":1},"tags":[{"location":{"line":1,"column":1},"name":"@foo","id":"4d691f08-00f3-4752-9efb-d735733e07d6"}],"language":"en","keyword":"Feature","name":"Basic Arithmetic","children":[{"scenario":{"location":{"line":4,"column":3},"keyword":"Scenario","name":"A Calculator","steps":[{"location":{"line":5,"column":5},"keyword":"Given ","text":"a calculator I just turned on","id":"e0157ff5-c589-4092-80f8-634d0c330874"}],"id":"f13f5e51-dd44-462b-bf87-7032d2f9b2ca"}}]}}}
{"pickle":{"id":"708c4b00-cbf4-4c94-9768-89ac044540d1","uri":"classpath:io/cucumber/examples/java/basic_arithmetic.feature","name":"A Calculator","language":"en","steps":[{"text":"a calculator I just turned on","id":"484f0e79-ee73-4205-a1c4-9eb87eea2fbd","astNodeIds":["e0157ff5-c589-4092-80f8-634d0c330874"]}],"tags":[{"name":"@foo","astNodeId":"4d691f08-00f3-4752-9efb-d735733e07d6"}],"astNodeIds":["f13f5e51-dd44-462b-bf87-7032d2f9b2ca"]}}
{"hook":{"id":"5cc9a09b-f425-4924-b539-2d1ccb47ca78"}}
{"stepDefinition":{"id":"7e1849c5-f19f-4aa7-bb4d-85ff79b8c5a2","pattern":{"source":"a calculator I just turned on"}}}
{"testCase":{"id":"8d16022b-6746-41b3-936e-cbd53dcd36d5","pickleId":"708c4b00-cbf4-4c94-9768-89ac044540d1","testSteps":[{"id":"5959fed3-3ffa-4969-8209-4bd27c3cd0da","hookId":"5cc9a09b-f425-4924-b539-2d1ccb47ca78"},{"id":"6adf9815-a8b5-4fd6-9d7e-5d0a0ad38b68","pickleStepId":"484f0e79-ee73-4205-a1c4-9eb87eea2fbd","stepDefinitionIds":["7e1849c5-f19f-4aa7-bb4d-85ff79b8c5a2"],"stepMatchArgumentsLists":[{}]}]}}
{"testCaseStarted":{"timestamp":{"seconds":"1593102300","nanos":738911000},"testCaseId":"8d16022b-6746-41b3-936e-cbd53dcd36d5","id":"51cd87af-1d1e-4644-bc37-f5a203cd3977"}}
{"testStepStarted":{"timestamp":{"seconds":"1593102300","nanos":740572000},"testStepId":"5959fed3-3ffa-4969-8209-4bd27c3cd0da","testCaseStartedId":"51cd87af-1d1e-4644-bc37-f5a203cd3977"}}
{"attachment":{"testStepId":"5959fed3-3ffa-4969-8209-4bd27c3cd0da","testCaseStartedId":"51cd87af-1d1e-4644-bc37-f5a203cd3977","body":"HELLLLOO FIRST","mediaType":"text/x.cucumber.log+plain"}}
{"testStepFinished":{"testStepResult":{"status":"PASSED","duration":{"nanos":2984000}},"timestamp":{"seconds":"1593102300","nanos":743556000},"testStepId":"5959fed3-3ffa-4969-8209-4bd27c3cd0da","testCaseStartedId":"51cd87af-1d1e-4644-bc37-f5a203cd3977"}}
{"testStepStarted":{"timestamp":{"seconds":"1593102300","nanos":749474000},"testStepId":"6adf9815-a8b5-4fd6-9d7e-5d0a0ad38b68","testCaseStartedId":"51cd87af-1d1e-4644-bc37-f5a203cd3977"}}
{"testStepFinished":{"testStepResult":{"status":"PASSED","duration":{"nanos":557000}},"timestamp":{"seconds":"1593102300","nanos":750031000},"testStepId":"6adf9815-a8b5-4fd6-9d7e-5d0a0ad38b68","testCaseStartedId":"51cd87af-1d1e-4644-bc37-f5a203cd3977"}}
{"testCaseFinished":{"timestamp":{"seconds":"1593102300","nanos":750588000},"testCaseStartedId":"51cd87af-1d1e-4644-bc37-f5a203cd3977"}}
{"testRunFinished":{"success":true,"timestamp":{"seconds":"1593102300","nanos":754104000}}}
```

**Comment by mpkorstanje** [comment]
Fixed in `cucumber-html:7.0.0`
